src/shogun/clustering/KMeans.cpp

```diff
@@ -236,11 +163,11 @@ bool CKMeans::train_machine(CFeatures* data)
 		CDenseFeatures<float64_t>::obtain_from_generic(distance->get_lhs());
 
 	ASSERT(lhs);
-	int32_t XSize=lhs->get_num_vectors();
+	int32_t lhs_size=lhs->get_num_vectors();
 	dimensions=lhs->get_num_features();
 	const int32_t XDimk=dimensions*k;
 
-	ASSERT(XSize>0 && dimensions>0);
+	ASSERT(lhs_size>0 && dimensions>0);
```

Comment:
can you make this a require, and also provide proper error messages

https://github.com/shogun-toolbox/shogun/wiki/Assertions